### PR TITLE
Grt: refactor critical nets percentage information

### PR DESCRIPTION
**Pull request by arthurjolo**
During the discussions on issue [#4121](https://github.com/The-OpenROAD-Project/OpenROAD/issues/4121) ,it was realized that the information of critical nets percentage was stored twice, some times creating a difference on the values stored. This branch makes the information of critical nets percentage to be stored only on fastroute.

### CODE REVIEW

**Comment by arthurjolo** [comment]
This branch has passed throw a secure-CI, public designs are OK, but private designs need to check by someone with access.

**Comment by maliberty** [comment]
You should get an email once the private designs finish with a summary.

**Comment by eder-matheus** [comment]
> This branch has passed throw a secure-CI, public designs are OK, but private designs need to be check by someone with access.

Could you share the secure- branch name? Also, fix clang-format.

**Comment by maliberty** [comment]
All designs that have finished so far have passed but a few big ones are still running.

**Comment by arthurjolo** [comment]
> > This branch has passed throw a secure-CI, public designs are OK, but private designs need to be check by someone with access.
> 
> Could you share the secure- branch name? Also, fix clang-format.

Just fixed the clang-format, and the branch name is: secure-grt_refactor_cn_percentage

File: src/grt/src/GlobalRouter.cpp [hunk 1]
```diff
@@ -187,12 +186,13 @@ std::vector<Net*> GlobalRouter::initFastRoute(int min_routing_layer,
   setCapacities(min_routing_layer, max_routing_layer);
 
   if (sta_->getDbNetwork()->defaultLibertyLibrary() == nullptr) {
-    critical_nets_percentage_ = 0;
     logger_->warn(
         GRT,
         300,
         "Timing is not available, setting critical nets percentage to 0");
```
Comment (eder-matheus) [inline src/grt/src/GlobalRouter.cpp:192]: Not related to your PR, but please add a period at the end of this sentence.

File: src/grt/src/GlobalRouter.cpp [hunk 2]
```diff
@@ -1365,7 +1320,14 @@ void GlobalRouter::setMaxLayerForClock(const int max_layer)
 
 void GlobalRouter::setCriticalNetsPercentage(float critical_nets_percentage)
 {
-  critical_nets_percentage_ = critical_nets_percentage;
+  if (sta_->getDbNetwork()->defaultLibertyLibrary() == nullptr) {
+    critical_nets_percentage = 0;
+    logger_->warn(
+        GRT,
+        301,
+        "Timing is not available, setting critical nets percentage to 0");
```
Comment (eder-matheus) [inline src/grt/src/GlobalRouter.cpp:1328]: Same here. Add the period at the end.

**Review by eder-matheus — CHANGES_REQUESTED** [review]
Just small requests, but this is almost ready to merge. I can see that secure-ci is almost done, and metrics are green for every design.

**Review by eder-matheus — APPROVED** [review]

**Comment by eder-matheus** [comment]
@arthurjolo The ok files of GRT unit tests need to be updated.

**Pull request merged**